tests/core/mempool/test_mempool_manager.py

```diff
@@ -329,3 +329,54 @@ async def test_sb_twice_with_eligible_coin_and_different_spends_order() -> None:
     assert result == (expected_cost, MempoolInclusionStatus.PENDING, Err.MEMPOOL_CONFLICT)
     assert mempool_manager.get_spendbundle(sb_name) == sb
     assert mempool_manager.get_spendbundle(reordered_sb_name) is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "opcode,lock_value,expected_status,expected_error",
+    [
+        (ConditionOpcode.ASSERT_SECONDS_RELATIVE, -2, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_SECONDS_RELATIVE, -1, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_SECONDS_RELATIVE, 0, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_SECONDS_RELATIVE, 1, MempoolInclusionStatus.FAILED, Err.ASSERT_SECONDS_RELATIVE_FAILED),
+        (ConditionOpcode.ASSERT_HEIGHT_RELATIVE, -2, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_HEIGHT_RELATIVE, -1, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_HEIGHT_RELATIVE, 0, MempoolInclusionStatus.PENDING, Err.ASSERT_HEIGHT_RELATIVE_FAILED),
+        (ConditionOpcode.ASSERT_HEIGHT_RELATIVE, 1, MempoolInclusionStatus.PENDING, Err.ASSERT_HEIGHT_RELATIVE_FAILED),
+        (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 4, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 5, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 6, MempoolInclusionStatus.PENDING, Err.ASSERT_HEIGHT_ABSOLUTE_FAILED),
+        (ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE, 7, MempoolInclusionStatus.PENDING, Err.ASSERT_HEIGHT_ABSOLUTE_FAILED),
+        (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, TEST_TIMESTAMP - 1, MempoolInclusionStatus.SUCCESS, None),
+        (ConditionOpcode.ASSERT_SECONDS_ABSOLUTE, TEST_TIMESTAMP, MempoolInclusionStatus.SUCCESS, None),
+        (
+            ConditionOpcode.ASSERT_SECONDS_ABSOLUTE,
+            TEST_TIMESTAMP + 1,
+            MempoolInclusionStatus.FAILED,
+            Err.ASSERT_SECONDS_ABSOLUTE_FAILED,
+        ),
+        (
+            ConditionOpcode.ASSERT_SECONDS_ABSOLUTE,
+            TEST_TIMESTAMP + 2,
+            MempoolInclusionStatus.FAILED,
+            Err.ASSERT_SECONDS_ABSOLUTE_FAILED,
+        ),
+    ],
+)
+async def test_ephemeral_timelock(
+    opcode: ConditionOpcode,
+    lock_value: int,
+    expected_status: MempoolInclusionStatus,
+    expected_error: Optional[Err],
+) -> None:
+    mempool_manager = await instantiate_mempool_manager(get_coin_record_for_test_coins)
```

Comment:
Absolute height 4 is OK, but height 3 not. As a reviewer of this test I can't tell whether that's right now not, There's nothing telling me what block height this mempool is at. Ideally, that height would be visible in this test and not require digging into abstractions to find it.

Reply:
Block height and timestamp are more explicit now.